Brokerages/Zerodha/ZerodhaBrokerage.cs

```diff
@@ -0,0 +1,1352 @@
+        public ZerodhaBrokerage(string tradingSegment, string zerodhaProductType, string apiKey, string apiSecret, IAlgorithm algorithm, IDataAggregator aggregator)
+            : base("Zerodha")
+        {
+            _tradingSegment = tradingSegment;
+            _zerodhaProductType = zerodhaProductType;
+            _algorithm = algorithm;
+            _aggregator = aggregator;
+            _kite = new Kite(apiKey, apiSecret);
+            _apiKey = apiKey;
+            _accessToken = apiSecret;
+            WebSocket = new ZerodhaWebSocketClientWrapper();
+            _wssUrl += string.Format(CultureInfo.InvariantCulture, "?api_key={0}&access_token={1}", _apiKey, _accessToken);
+            WebSocket.Initialize(_wssUrl);
+            WebSocket.Message += OnMessage;
+            WebSocket.Open += (sender, args) =>
+            {
+                Log.Trace($"ZerodhaBrokerage(): WebSocket.Open. Subscribing");
+                Subscribe(GetSubscribed());
+            };
+            WebSocket.Error += OnError;
+            _symbolMapper = new ZerodhaSymbolMapper(_kite);
+
+            var subscriptionManager = new EventBasedDataQueueHandlerSubscriptionManager();
+            subscriptionManager.SubscribeImpl += (s, t) =>
+            {
+                Subscribe(s);
+                return true;
+            };
+            subscriptionManager.UnsubscribeImpl += (s, t) => Unsubscribe(s);
+            SubscriptionManager = subscriptionManager;
+
+            Log.Trace("Start Zerodha Brokerage");
+        }
+
+        /// <summary>
+        /// Subscribes to the requested symbols (using an individual streaming channel)
+        /// </summary>
+        /// <param name="symbols">The list of symbols to subscribe</param>
+        public void Subscribe(IEnumerable<Symbol> symbols)
+        {
+            if (symbols.Count() <= 0)
+            {
+                return;
+            }
+            foreach (var symbol in symbols)
+            {
+                var market = GetSymbolMarket(symbol);
+                var instrumentToken = _symbolMapper.GetZerodhaInstrumentToken(symbol.ID.Symbol, market);
+                if (instrumentToken == 0)
+                {
+                    Log.Error("Invalid Zerodha Instrument token");
+                }
+                if (!subscribeInstrumentTokens.Contains(instrumentToken.ToStringInvariant()))
+                {
+                    subscribeInstrumentTokens.Add(instrumentToken.ToStringInvariant());
+                    unSubscribeInstrumentTokens.Remove(instrumentToken.ToStringInvariant());
+                    _subscriptionsById[instrumentToken.ToStringInvariant()] = symbol;
+                }
+            }
+            string request = "{\"a\":\"subscribe\",\"v\":[" + String.Join(",", subscribeInstrumentTokens.ToArray()) + "]}";
+            string requestFullMode = "{\"a\":\"mode\",\"v\":[\"full\",[" + String.Join(",", subscribeInstrumentTokens.ToArray()) + "]]}";
```

Comment:
nit - string -> var

What's `request` vs `requestFullMode`? Worth a comment I think 👍 
We do not unsubscribe of the full mode?